### PR TITLE
Change gtag events to track dialog to purchase

### PR DESCRIPTION
**Pull request by BeEdward**

### CODE REVIEW

File: templates/ui.html [hunk 1]
```diff
@@ -1177,17 +1177,111 @@
...
+              }
+            )
+            .then(response => {
+              if (response.status === 200) {
+                return response.json();
+              }
+              return Promise.reject(response);
+            })
+            .then((data) => {
+              if (typeof callback === 'function') {
```
Comment (linjiX) [inline templates/ui.html:1268]: put callback in `finally` block?

File: templates/ui.html [hunk 2]
```diff
@@ -1177,17 +1177,111 @@
...
               URL: "/user#/subscription?type=subscription",
             },
+            analyticsProductInfo: {
+              itemsMapping: {},
+              itemsInLists: {},
+            },
           };
         },
         methods: {
+          getItemsMapping(callback = null) {
```
Comment (linjiX) [inline templates/ui.html:1213]: this function is not be called?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

File: templates/ui.html [hunk 3]
```diff
@@ -1177,17 +1177,110 @@
...
+            .finally(() => {
+              if (typeof callback === 'function') {
+                callback();
+              }
+            });
+          },
+          getItemUniqueId(listID) {
+            for (let key in this.analyticsProductInfo.itemsMapping) {
+              if (listID.includes(key)) {
+                return this.analyticsProductInfo.itemsMappingitemsMapping[key];
```
Comment (linjiX) [inline templates/ui.html]: should `itemsMappingitemsMapping` be `itemsMapping`?

File: templates/ui.html [hunk 4]
```diff
@@ -1177,17 +1177,110 @@
...
+              }
+            }
+            return null;
+          },
           addPopupGtagEvent() {
-            gtag("event", "view_item", {
-              items: [
+            const listUniqueId = this.getItemUniqueId(this.popup.URL) || this.popup.URL;
```
Comment (linjiX) [inline templates/ui.html:1282]: this code block to get `listUniqueId`, `domain` and `items` seems appears twice, should it be a function?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Comment by BeEdward** [comment]
cherry-picked to diffus

**Comment by BeEdward** [comment]
cherry-picked to diffus

**Pull request closed without merging**